diff --git a/dist/index.js b/dist/index.js
index 7d82df112a40b76e795d02b43e9c210b842e0297..a839205abbfbc62f7deb2a5a211d11dd03df757d 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -116,7 +116,7 @@ class Compiler {
         const name = this.getName(node.name);
         const params = node.parameters.map(this.compileNode, this);
         const items = [this.compileOptType(node.type)].concat(params);
-        return `"${name}": t.func(${items.join(", ")})`;
+        return `"${name}": t.opt(t.func(${items.join(", ")}))`;
     }
     _compileTypeReferenceNode(node) {
         if (!node.typeArguments) {
@@ -182,6 +182,7 @@ class Compiler {
         return `t.opt(${this.compileNode(node.type)})`;
     }
     _compileEnumDeclaration(node) {
+        return ''
         const name = this.getName(node.name);
         const members = node.members.map(m => `  "${this.getName(m.name)}": ${getTextOfConstantValue(this.checker.getConstantValue(m))},\n`);
         this.exportedNames.push(name);
